### PR TITLE
Expose AnalysisId for further use

### PR DESCRIPTION
**Pull request by hangar18rip**
I have added an option to expose the AnalysisId for further usage

### CODE REVIEW

**Comment by bgavrilMS** [comment]
Hi @hangar18rip 

We're happy to publish the analysis Id as an extensibility point, but I have a few asks: 

1. Could you tell me a bit more about the scenario here? Why do you need this?
2. I don't think we should change the task UI. Instead, we should just publish the analysis ID every time. It has no user impact. 
3. Currently, you'd only get the analysis ID if the build breaker is enabled. For this to be a proper extensibility point, we should make it available every time. In other words, we should decouple fetching the analysis ID from the build breaker logic. 
4. This needs a unit test. See https://github.com/Microsoft/vso-agent-tasks/blob/master/docs/runningtests.md for details. 

Feel free to answer just question 1 though, I'm happy to do 2-4 if you wish.

**Comment by hangar18rip** [comment]
Hi,


1.       I would like to display more information about the analysis in the build summary. Ex : display the detail of the quality gate status.

2.       From my understanding of SonarQube (I might be wrong), the asynchronous processing of the metrics makes the Analysis ID only available after a certain amout of time right ? So the AnalysisID might be “optional” from my point of view.

3.       This is the reason why I have made this modification. I reuse the BuildBreaker logic to get the AnalysisID but, once again, I might be wrong. And exploiting the AnalysisID would make more sens if the processing has ended.

4.       I will add more testing

Regards,

De : Bogdan Gavril [mailto:notifications@github.com]
Envoyé : lundi 4 avril 2016 10:06
À : Microsoft/vso-agent-tasks <vso-agent-tasks@noreply.github.com>
Cc : Olivier Delmotte <olivier@odelmotte.fr>
Objet : Re: [Microsoft/vso-agent-tasks] Expose AnalysisId for further use (#1473)


Hi @hangar18rip<https://github.com/hangar18rip>

We're happy to publish the analysis Id as an extensibility point, but I have a few asks:

  1.  Could you tell me a bit more about the scenario here? Why do you need this?
  2.  I don't think we should change the task UI. Instead, we should just publish the analysis ID every time. It has no user impact.
  3.  Currently, you'd only get the analysis ID if the build breaker is enabled. For this to be a proper extensibility point, we should make it available every time. In other words, we should decouple fetching the analysis ID from the build breaker logic.
  4.  This needs a unit test. See https://github.com/Microsoft/vso-agent-tasks/blob/master/docs/runningtests.md for details.

Feel free to answer just question 1 though, I'm happy to do 2-4 if you wish.

**Comment by bgavrilMS** [comment]
Ah, we're actually planning to improve the build summary. The work has been spec-ed and is tracked by this story: https://jira.sonarsource.com/browse/MMF-74. I'll add the spec to the ticket after it has been internally reviewed. I'm likely to start work this week on this. 

We're thinking that the new build summary should have: 

- the breakdown of product / test / skipped projects (like now)
- the quality gate status
- the tech debt numbers (the Sqale rating, new debt etc.)

Any feedback on this? Would you expect other data to be available?

In conclusion, I'd like to propose we close this issue and wait for the story to be implemented.

**Comment by hangar18rip** [comment]
Fine.
I would just had the violated gates with values to have a quick view (with a red visual ideally).
It is ok for me to reject this pull request

**Comment by bgavrilMS** [comment]
Hi again @hangar18rip, 

I've merged the new summary report, or at least the first part of it. It will now display the Quality Gate. 

Given that it will take a few weeks (around 4 I think) before the changes become live, you may want to build the 2 SonarQube tasks and upload them to your account to get the feature earlier. Let me know if you need help with that.
